### ISSUE TITLE
`match_overlapping_arm` false negative when overlap after included range

### ISSUE THREAD

**Issue by mikerite** [rust-lang/rust-clippy#7816]
Lint name: match_overlapping_arm

I tried this code:

```rust
#![warn(clippy::match_overlapping_arm)]

fn main() {
    let x = 10usize;
    match x {
        5..=10 => println!("a"),
        0..=20 => println!("b"),
        21..=30 => println!("c"),
        21..=40 => println!("d"),
        _ => println!("wild card")
    }

    let x = 10usize;
    match x {
        21..=30 => println!("c"),
        21..=40 => println!("d"),
        _ => println!("wild card")
    }
}

```

I expected to see this happen: The range `21..=30` would be linted in both match expressiona but it's only done in the second one.

Instead, this happened: Only the second match expression was linted.

```
 Checking playground v0.0.1 (/playground)
warning: some ranges overlap
  --> src/main.rs:15:9
   |
15 |         21..=30 => println!("c"),
   |         ^^^^^^^
   |
note: the lint level is defined here
  --> src/main.rs:1:9
   |
1  | #![warn(clippy::match_overlapping_arm)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
note: overlaps with this
  --> src/main.rs:16:9
   |
16 |         21..=40 => println!("d"),
   |         ^^^^^^^
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#match_overlapping_arm

warning: `playground` (bin "playground") generated 1 warning
    Finished dev [unoptimized + debuginfo] target(s) in 0.56s
```
[playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=7838f01fa8c5e130c9c4790c0a41d258)
